framework/test/mocha/unit/modules/http_api/http_api.js

```diff
@@ -1,112 +1,226 @@
 /* eslint-disable mocha/no-pending-tests */
+const rewire = require('rewire');
+
+const HttpApi = rewire('../../../../../src/modules/http_api/http_api');
 
 describe('HttpApi', () => {
-	it('should export HttpApi class');
+	let httpApi;
+	const stubs = {};
+	const loggerConfig = {
+		logFileName: 'logs.log',
+	};
+	const cacheConfig = 'aCacheConfig';
+	const systemConfig = 'aSystemConfig';
+	const storageConfig = {
+		logFileName: 'logs.log',
+	};
+
+	beforeEach(async () => {
+		stubs.channel = {
+			invoke: sinonSandbox.stub(),
+		};
+		stubs.options = {
+			config: sinonSandbox.stub(),
+			constants: {
+				ACTIVE_DELEGATES: 101,
+			},
+		};
+		stubs.logger = {
+			debug: sinonSandbox.stub(),
+		};
+		stubs.system = sinonSandbox.stub();
+		stubs.storage = sinonSandbox.stub();
+		stubs.cache = sinonSandbox.stub();
+		stubs.servers = {
+			expressApp: sinonSandbox.stub(),
+			httpServer: sinonSandbox.stub(),
+			httpsServer: sinonSandbox.stub(),
+			wsServer: sinonSandbox.stub(),
+			wssServer: sinonSandbox.stub(),
+		};
+
+		stubs.createLoggerComponent = sinonSandbox.stub().returns(stubs.logger);
+		stubs.createSystemComponent = sinonSandbox.stub().returns(stubs.system);
+		stubs.createCacheComponent = sinonSandbox.stub().returns(stubs.cache);
+		stubs.createStorageComponent = sinonSandbox.stub().returns(stubs.storage);
+		stubs.bootstrapCache = sinonSandbox.stub();
+		stubs.bootstrapStorage = sinonSandbox.stub();
+		stubs.setupServers = sinonSandbox.stub().resolves(stubs.servers);
+		stubs.bootstrapSwagger = sinonSandbox.stub();
+		stubs.startListening = sinonSandbox.stub();
+		stubs.subscribeToEvents = sinonSandbox.stub();
+
+		stubs.channel.invoke
+			.withArgs('lisk:getComponentConfig', 'logger')
+			.resolves(loggerConfig);
+		stubs.channel.invoke
+			.withArgs('lisk:getComponentConfig', 'storage')
+			.resolves(storageConfig);
+		stubs.channel.invoke
+			.withArgs('lisk:getComponentConfig', 'cache')
+			.resolves(cacheConfig);
+		stubs.channel.invoke
+			.withArgs('lisk:getComponentConfig', 'system')
+			.resolves(systemConfig);
+
+		HttpApi.__set__('createLoggerComponent', stubs.createLoggerComponent);
```

Comment:
Instead of using `rewire`, you could just stub the original function? for example, 
```
const loggerComponent = require('../../../xxx/component/logger');
sinonSandbox.stub(loggerComponent, 'createLoggerComponent').returns(stubs.logger);
```
Apply to other rewire related stuff too